src/test/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/adhoc/createtasks/CreateTaskJobOutcomeServiceTest.java

```diff
@@ -0,0 +1,134 @@
+
+    @SneakyThrows
+    private void changePrivateConstant(Object instance, String constantName, int newValue) {
+        Field field = instance.getClass().getDeclaredField(constantName);
+        field.setAccessible(true);
+
+        Field modifiers = field.getClass().getDeclaredField("modifiers");
+        modifiers.setAccessible(true);
+        modifiers.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+
+        field.set(instance, newValue);
+    }
+
+    @Test
+    void givenCamundaClientThrowsExceptionShouldGetJobOutcome() {
+        when(camundaClient.getTasksByTaskVariables(
+            eq(SOME_SERVICE_TOKEN),
+            eq("caseId_eq_someCaseId,taskType_eq_reviewAppealSkeletonArgument"),
+            eq("created"),
+            eq("desc")
+        )).thenThrow(new RuntimeException("some exception"));
+
+        CreateTaskJobOutcome actual = createTaskJobOutcomeService.getJobOutcome(
+            SOME_SERVICE_TOKEN,
+            SOME_CASE_ID
+        );
+
+        assertThat(actual).isEqualTo(CreateTaskJobOutcome.builder()
+                                         .caseId(SOME_CASE_ID)
+                                         .created(false)
+                                         .build());
+    }
+
+    @ParameterizedTest
+    @MethodSource("scenarioProvider")
+    void shouldGetJobOutcome(List<CamundaTask> camundaTaskList, CreateTaskJobOutcome expectedOutcome) {
+        when(camundaClient.getTasksByTaskVariables(
+            eq(SOME_SERVICE_TOKEN),
+            eq("caseId_eq_someCaseId,taskType_eq_reviewAppealSkeletonArgument"),
+            eq("created"),
+            eq("desc")
+        )).thenReturn(camundaTaskList);
+
+        CreateTaskJobOutcome actual = createTaskJobOutcomeService.getJobOutcome(
+            SOME_SERVICE_TOKEN,
+            SOME_CASE_ID
+        );
+
+        assertThat(actual).isEqualTo(expectedOutcome);
+    }
+
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
+    private static Stream<Arguments> scenarioProvider() {
+        Arguments taskIsCreatedScenario = Arguments.of(
+            List.of(new CamundaTask(
+                "some task id",
+                "Review Appeal Skeleton Argument",
+                "some process instance id"
+            )),
+            CreateTaskJobOutcome.builder()
+                .taskId("some task id")
+                .processInstanceId("some process instance id")
+                .caseId(SOME_CASE_ID)
+                .created(true)
+                .build()
+        );
+
+        Arguments taskIsNotFoundScenario = Arguments.of(
+            Collections.emptyList(),
+            CreateTaskJobOutcome.builder()
+                .caseId(SOME_CASE_ID)
+                .created(false)
+                .build()
+        );
+
+        Arguments camundaTaskListIsNullScenario = Arguments.of(
+            null,
+            CreateTaskJobOutcome.builder()
+                .caseId(SOME_CASE_ID)
+                .created(false)
+                .build()
+        );
+
```

Comment:
As per sonar -> One more edge case test to cover camundaList !empty() Or !=Null And Task.getName().equals("Review Appeal Skeleton Argument") ?

Reply:
done